tests/pytests/unit/utils/test_msgpack.py

```diff
@@ -0,0 +1,69 @@
+import os
+
+import pytest
+import salt.utils.msgpack
+from tests.support.mock import MagicMock, patch
+from tests.support.runtests import RUNTIME_VARS
+
+
+@pytest.fixture
+def test_fn():
+    fn_ = os.path.join(RUNTIME_VARS.TMP, "test_msgpack")
+    yield fn_
+    if os.path.exists(fn_):
+        os.remove(fn_)
+
+
+def test_load_encoding(test_fn):
```

Comment:
If you don't need anything special, you can replace your `test_fn` with https://docs.pytest.org/en/stable/tmpdir.html#the-tmpdir-fixture

Also, don't name your fixtures `test_`, even though now pytest apparently does not complain or blowup, likely because it's decorated, that naming convention, suggests, it's a test, while it's now. It's just so it reads better...

Reply:
I see the advantage of using the custom fixture to be that i only have to define my file path once. If I use the `tmp_path` fixture I would need to define this path name twice in each test method. Is there a better way to be able to do that? I will definitely change the name of the fixture, thanks

Reply:
I guess i dont really need the same filename, so i'll just update the tests to use that fixture. thanks

Reply:
Glad you sorted that out.
You could also use the `tmpdir` fixture on your fixture, the advantage is that it cleans itself up, we don't have to.